content/blogposts/2022/code-intelligence-platform.md

```diff
@@ -0,0 +1,33 @@
+---
+title: "From code search toward a code intelligence platform"
+publishDate: 
+description: TODO
+tags: [blog]
+slug: "code-search-to-code-intelligence"
+published: false
+heroImage: 
+socialImage: 
+changelogItems:
+---
+
+> NOTE: This post is very WIP. To be edited and tightened up a lot.
+
+No matter how good your other dev tools, coding is still complex. Even if you've configured your editor just right, sped up your test suite, set up continuous deployment, added observability, etc., coding is still complex. Higher-level languages, libraries, frameworks, cloud providers, etc., that encapsulate complexity just let you spend your complexity budget elsewhere. The overall complexity of code has increased, not declined, over time, and there's no sign of that trend changing.
+
+As consumers of software, we (mostly) benefit from this phenomenon in the form of better technology in our everyday lives.
+
+But as developers of software, we all feel the pain of code complexity: the struggle to understand big codebases, not being able to get in flow, encountering tech debt that turns a tiny fix into a multi-day ordeal, all the wasted time, etc.
+
+Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you find things *about* code? Let's call it a code intelligence platform. Compared to code search, it's more helpful, more times per day. This is like how Google went from helping you find things *on* the web (matching web pages), to helping you find things *from* the web (information and answers). Here are some examples `TODO: make it clear which these are available and which are future directions`:
```

Comment:
```suggestion
Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you [understand/learn] things *about* code? Let's call it a code intelligence platform. Compared to code search, it's more helpful, more times per day. This is like how Google went from helping you find things *on* the web (matching web pages), to helping you [understand/learn] things *from* the web (information and answers). Here are some examples `TODO: make it clear which these are available and which are future directions`:
```

@sqs - I like the contrast of "find" vs "[understand/learn]" in addition to the contrast of "in" vs "about" code. Understanding or learning is core to code intelligence and more powerful than finding.

Reply:
```suggestion
Code search is helpful in tackling complexity, but it's not enough. Code search is for finding things *in* code (matching lines in files). What helps you learn things *about* code? Let's call it a code intelligence platform. Compared to code search, it's more helpful, more times per day. This is like how Google went from helping you find things *on* the web (matching web pages), to helping you learn things *from* the web (information and answers). Here are some examples:
```